test/test_OUT_4.f

```diff
@@ -0,0 +1,153 @@
+
+C*      The following will ensure that subroutine STNDRD is called
+C*      internally during the subsequent calls to WRITSB and CLOSMG.
+
+        CALL STDMSG ('Y')
+        print *, '        STDMSG'
+
+C*      Append a (tank) receipt time to Section 1 of each output message
+
+	CALL STRCPT ( 'Y', 2020, 11, 4, 15, 29 )
+	print *, '        STRCPT'
+
+C*	Process 1 message with 1 subset from infile1.
+
+        CALL READMG ( 11, cmgtag, imgdt, iermg )
+	print *, '        READMG'
+
+        IF ( iermg .eq. 0 ) THEN
+          CALL READSB ( 11, iersb )
+	  print *, '        READSB'
+
+          IF ( iersb .eq. 0 ) THEN
+	    CALL OPENMB ( 13, 'NC007000', 2020022514 )
+	    print *, '        OPENMB'
+
+            CALL UFBSEQ ( 11, r8arr1, MXVAL1, 1, nlv, 'NC007000' )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'NC007000' )
+	    print *, '        UFBSEQ'
+
+            CALL WRITSB ( 13 )
+          END IF
+        END IF
+
+	CALL CLOSMG ( 13 )
+	print *, '        CLOSMG'
+
+C*	Process 1 message with 4 subset from infile2.
+
+        CALL READMG ( 12, cmgtag, imgdt, iermg )
+
+C*      Turn off output message standardization.
+
+        CALL STDMSG ('N')
+        
+C*      Write DX table information for this message into the
+C*      output file.
+
+        CALL WRDXTB ( 12, 13 )
+	print *, '        WRDXTB'
+
+        IF ( iermg .eq. 0 ) THEN
+
+          nsub = 0
+
+          DO WHILE ( IREADSB ( 12 ) .eq. 0 )
+
+            nsub = nsub + 1
+
+            CALL UFBSEQ ( 12, r8arr1, MXVAL1, 1, nlv, 'DATETMLN' )
+            ibfdt = ( IDNINT(r8arr1(2)) * 1000000 ) +
+     +              ( IDNINT(r8arr1(3)) * 10000 ) +
+     +              ( IDNINT(r8arr1(4)) * 100 ) + IDNINT(r8arr1(5))
+	    CALL OPENMB ( 13, 'MSTTB001', ibfdt )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'DATETMLN' )
+
+            WRITE ( UNIT = smid, FMT = '(A,I1.1)' ) 'STATION#', nsub
+            IF ( nsub .eq. 1 ) THEN
+              CALL READLC ( 12, dummystr, 'DUMMYSTR' )
+              IF ( ICBFMS( dummystr, 9 ) .eq. 0 ) smid = dummystr
+            END IF
+
+            CALL UFBSEQ ( 12, r8arr1, MXVAL1, 1, nlv, 'IDLSIPTM' )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'IDLSIPTM' )
+            CALL UFBSEQ ( 12, r8arr1, MXVAL1, 1, nlv, 'HAVCOLS' )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'HAVCOLS' )
+            CALL UFBSEQ ( 12, r8arr1, MXVAL1, 1, nlv, 'CLINRVSD' )
+            CALL UFBSEQ ( 13, r8arr1, MXVAL1, 1, nlv, 'CLINRVSD' )
+            CALL UFBSEQ ( 12, r8arr2, MXVAL2, MXLVL, nlv2, 'TDWPRAOB' )
+
+            CALL DRFINI ( 13, nlv2, 1, '(TDWPRAOB)' )
+            CALL UFBSEQ ( 13, r8arr2, MXVAL2, nlv2, nlv, 'TDWPRAOB' )
+
+            CALL HOLD4WLC ( 13, smid, 'SMID' )
+            CALL WRITSA ( 13, MXBFMG, mgbf, lmgbf )
+            IF ( nsub .eq. 1 ) THEN
+              CALL WRITLC ( 13, dummystr, 'DUMMYSTR' )
+            END IF
+           
+          END DO
+
+          CALL WRITSA ( -13, MXBFMG, mgbf, lmgbf )
```

Comment:
You seem to be writing a file in testing, but  don't see where the file is opened and checked for correctness...

Reply:
For all of these output tests, the correctness of the output is verified within test_wrapper.sh, where the ${outname}.bufr file is checked for bitwise equality vs. testfiles/${refname} using "cmp -s".  So the generated file is indeed opened and checked for correctness.